src/components/application_manager/include/application_manager/application_manager_impl.h

```diff
@@ -1499,6 +1552,12 @@ class ApplicationManagerImpl
 
   volatile bool is_stopping_;
 
+#ifdef BUILD_TESTS
+  // Methods for test usage
+ public:
+  void AddMockApplication(ApplicationSharedPtr mock_app);
```

Comment:
Sure, added in https://github.com/smartdevicelink/sdl_core/pull/1700/commits/f5fb4e9ad9d6a4bfcec512ea7173fdc39f1bdd86.